src/mgr/MgrCommands.h

```diff
@@ -6,172 +6,175 @@
-	"pg", "rw", "cli,rest")
+	"pg", "rw")
 COMMAND("pg force-backfill name=pgid,type=CephPgid,n=N", "force backfill of <pgid> first", \
-	"pg", "rw", "cli,rest")
+	"pg", "rw")
 COMMAND("pg cancel-force-recovery name=pgid,type=CephPgid,n=N", "restore normal recovery priority of <pgid>", \
-	"pg", "rw", "cli,rest")
+	"pg", "rw")
 COMMAND("pg cancel-force-backfill name=pgid,type=CephPgid,n=N", "restore normal backfill priority of <pgid>", \
-	"pg", "rw", "cli,rest")
+	"pg", "rw")
 
 // stuff in osd namespace
 COMMAND("osd perf", \
         "print dump of OSD perf summary stats", \
         "osd", \
-        "r", \
-        "cli,rest")
+        "r")
 COMMAND("osd df " \
 	"name=output_method,type=CephChoices,strings=plain|tree,req=false", \
-	"show OSD utilization", "osd", "r", "cli,rest")
+	"show OSD utilization", "osd", "r")
 COMMAND("osd blocked-by", \
 	"print histogram of which OSDs are blocking their peers", \
-	"osd", "r", "cli,rest")
+	"osd", "r")
 COMMAND("osd pool stats " \
         "name=pool_name,type=CephPoolname,req=false",
         "obtain stats from all pools, or from specified pool",
-        "osd", "r", "cli,rest")
+        "osd", "r")
 COMMAND("osd reweight-by-utilization " \
 	"name=oload,type=CephInt,req=false " \
 	"name=max_change,type=CephFloat,req=false "			\
 	"name=max_osds,type=CephInt,req=false "			\
 	"name=no_increasing,type=CephChoices,strings=--no-increasing,req=false",\
 	"reweight OSDs by utilization [overload-percentage-for-consideration, default 120]", \
-	"osd", "rw", "cli,rest")
+	"osd", "rw")
 COMMAND("osd test-reweight-by-utilization " \
 	"name=oload,type=CephInt,req=false " \
 	"name=max_change,type=CephFloat,req=false "			\
 	"name=max_osds,type=CephInt,req=false "			\
-	"name=no_increasing,type=CephChoices,strings=--no-increasing,req=false",\
+	"name=no_increasing,type=CephBool,req=false",\
 	"dry run of reweight OSDs by utilization [overload-percentage-for-consideration, default 120]", \
-	"osd", "r", "cli,rest")
+	"osd", "r")
 COMMAND("osd reweight-by-pg " \
 	"name=oload,type=CephInt,req=false " \
 	"name=max_change,type=CephFloat,req=false "			\
 	"name=max_osds,type=CephInt,req=false "			\
 	"name=pools,type=CephPoolname,n=N,req=false",			\
 	"reweight OSDs by PG distribution [overload-percentage-for-consideration, default 120]", \
-	"osd", "rw", "cli,rest")
+	"osd", "rw")
 COMMAND("osd test-reweight-by-pg " \
 	"name=oload,type=CephInt,req=false " \
 	"name=max_change,type=CephFloat,req=false "			\
 	"name=max_osds,type=CephInt,req=false "			\
 	"name=pools,type=CephPoolname,n=N,req=false",			\
 	"dry run of reweight OSDs by PG distribution [overload-percentage-for-consideration, default 120]", \
-	"osd", "r", "cli,rest")
+	"osd", "r")
 
 COMMAND("osd destroy "	    \
         "name=id,type=CephOsdName " \
-        "name=sure,type=CephString,req=False",
+	"name=force,type=CephBool,req=false "
+        // backward compat synonym for --force
+	"name=yes_i_really_mean_it,type=CephBool,req=false", \
```

Comment:
What if, instead, we simply s/sure/force/, and s/CephString/CephBool/.  Then the compat code would only need to rewrite CephBool to CephString in the general case, and the code for each of these commands in the mon would need to tolerate getting either a boolean or a "--yes-i-really-mean-it" style string (it already does the latter).  I think that would make it so that the librados test case code would only need s/sure/force/?  And we would end up with only one field defined here in MonCommands.  We would lose the CephChoices that would tell you the magic force argument on older clients in ceph -h, but that is small and maybe even a good thing anyway?

Reply:
Oh, I see... this way lets you still do --yes-i-really-mean-it on the new code while also making it a non-positional flag.

In that case, I think the checks in the mon need to be adjusted so that they tolerate a string being passed for the compat field.  Probably a helper like bool cmdmap_compat_confirm(cmdmap, "force", "yes_i_really_mean_it") or something would make this easier and capture all of the compat hackery.

Reply:
This particular command definition already had both --force and --yes-i-really-mean-it, the two lines here are just synonym flags rather than anything related to CephBool vs. CephChoices